### DISCUSSION TITLE
How can we override iOS's UIApplicationDelegate events with Uno's project structure?

### DISCUSSION THREAD

**Discussion by abenadar** [unoplatform/uno#10663] [Q&A] [answered]
I want to run some background tasks when the application is moved to the background through overriding `UIApplicationDelegate.DidEnterBackground(UIApplication)`.

In Xamarin.iOS:

```
public class Application 
{
    static void Main(string[] args)
    {
        UIApplication.Main(args, null, "AppDelegate");
    }
}

[Register ("AppDelegate")]
public partial class AppDelegate : UIApplicationDelegate 
{ 
    public override void DidEnterBackground (UIApplication application) {
         nint taskID = UIApplication.SharedApplication.BeginBackgroundTask( () => {});
         new Task ( () => {
             UIApplication.SharedApplication.EndBackgroundTask(taskID);
         }).Start();
     }
}
```

Replacing this delegate with Uno's App class removes the inheritance, while making the default app inherit from UiApplication  naturally breaks cross-platform availability.

Is there a recommended approach for this?

**Comment by jeromelaban** [comment, marked as answer]
The `App.xaml.cs` `App` class inherits from `UIApplicationDelegate` on iOS, you can [use platform specific code](https://platform.uno/docs/articles/platform-specific-csharp.html) to place it in the App.xaml.cs.
